### PR TITLE
Refactoring, Sorting & Filtering

### PR DESCRIPTION
**Pull request by dride-for-sure**
- Refactoring
- Add sorting & filtering as well as improved pagination
- Add WIP school details page

### CODE REVIEW

**Review by Roman-devs — COMMENTED** [review]
Just a few comments since the overall level of syntax exceeds my understanding :)

File: frontend/src/common/indexData.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { v4 as uuid } from 'uuid';
+
+export const addIndizesToNestedData = (data, idChain) => {
```
[the remark this answers is not in the document]
Comment (dride-for-sure) [reply frontend/src/common/indexData.js:3]: Absolutely right! Thanks.

File: frontend/src/common/indexData.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  const id = idChain ? `${idChain}_${uuid()}` : uuid();
+  let indexedData = data;
+  if ((Object.prototype.toString.call(indexedData) === '[object Object]') && indexedData !== null) {
+    indexedData = { ...indexedData, id };
+    Object.keys(indexedData).forEach((key) => {
+      indexedData[key] = addIndizesToNestedData(indexedData[key], id);
+    });
+  }
+  if (Array.isArray(indexedData) && indexedData.length) {
```
[the remark this answers is not in the document]
Comment (dride-for-sure) [reply frontend/src/common/indexData.js:12]: Ref: Zoom Discussion.

File: frontend/src/common/searchParams.js [hunk 3]
```diff
@@ -1,8 +1,52 @@
...
 });
 
-export default getSearchParams;
+export const getQueryStringForPaginate = (whichWay, search, totalPages) => {
+  const params = getSearchParams(search);
+  const getPage = () => {
+    if (whichWay === 'last') {
+      return (Object.is(params.page, NaN) || params.page <= 1)
+        ? 1
+        : params.page - 1;
```
[the remark this answers is not in the document]
Comment (dride-for-sure) [reply frontend/src/common/searchParams.js:14]: Ref: Zoom-Discussion

File: frontend/src/components/assemblies/Paragraph.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ export default function Paragraph({ component, onChange }) {
       <Label>{component.type}</Label>
       <Textarea
         id="content"
+        placeholder="Enter the content..."
```
[the remark this answers is not in the document]
Comment (dride-for-sure) [reply frontend/src/components/assemblies/Paragraph.js:11]: Ref: Zoom-Discussion

File: frontend/src/components/parts/cms/EditSchool/SchoolProperty.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+export default function SchoolProperty({ property, onChange }) {
+  return (
+    <>
+      <Label htmlFor={property.name}>{property.name}</Label>
+      <Input
+        align="left"
+        id={property.name}
+        type="text"
+        placeholder={`Please enter ${property.name}`}
```
[the remark this answers is not in the document]
Comment (dride-for-sure) [reply frontend/src/components/parts/cms/EditSchool/SchoolProperty.js:13]: Ref: Zoom-Discussion

File: frontend/src/components/parts/cms/EditSchool/SchoolImage.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import CustomImage from '../../../image/Image';
+
+export default function SchoolImage({ imageUrl, onFileUpload, onFileDelete }) {
+  const uniqueFieldId = uuid();
+  return (
+    <>
+      <Label>School Image</Label>
+      <GridTwoOne>
+        <Left>
+          {imageUrl && <CustomImage src={imageUrl} alt="" />}
```
[the remark this answers is not in the document]
Comment (dride-for-sure) [reply frontend/src/components/parts/cms/EditSchool/SchoolImage.js:18]: Point :)

**Comment by dride-for-sure** [comment]
Merci @Roman-devs ! I appreciate it.

**Pull request merged**